Products/CMFPlone/browser/favicon.py

```diff
@@ -0,0 +1,26 @@
+from Products.CMFPlone.interfaces import ISiteSchema
+from plone.formwidget.namedfile.converter import b64decode_file
+from plone.namedfile.browser import Download
+from plone.namedfile.file import NamedImage
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+from plone.memoize import ram
+
+
+class SiteFavicon(Download):
+
+    def __init__(self, context, request):
+        super().__init__(context, request)
+        self.filename = None
+        self.data = None
+
+        registry = getUtility(IRegistry)
+        settings = registry.forInterface(ISiteSchema, prefix="plone")
+        if getattr(settings, 'site_favicon', False):
+            filename, data = b64decode_file(settings.site_favicon)
+            data = NamedImage(data=data, filename=filename)
+            self.data = data
+            self.filename = filename
+
+    def _getFile(self):  # todo add caching
```

Comment:
The caching is important.

Reply:
it looks like that im wrong and there is no caching needet. 
Why no caching ?
- If there is no favicon this view will not called
- Fallback is the Plone-default-favico at .../Plonesite/favicon.ico

Reply:
it looks like that im wrong and there is no caching needet in this case.